agent/fw_wordpress.c

```diff
@@ -569,16 +615,60 @@ NR_PHP_WRAPPER(nr_wordpress_apply_filters) {
       NR_PHP_WRAPPER_CALL;
     }
 
+    zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
     nr_wordpress_name_the_wt(tag, retval_ptr TSRMLS_CC);
   } else {
     NR_PHP_WRAPPER_CALL;
   }
+#endif //OAPI
+
+  nr_php_arg_release(&tag);
+}
+NR_PHP_WRAPPER_END
+
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+NR_PHP_WRAPPER(nr_wordpress_handle_tag_stack_after) {
+  (void)wraprec;
+  char* cleaned_tag = nr_stack_pop(&NRPRG(wordpress_tags));
+  nr_free(cleaned_tag);
+}
+NR_PHP_WRAPPER_END
 
+NR_PHP_WRAPPER(nr_wordpress_apply_filters_after) {
+  zval* tag = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (1 == nr_php_is_zval_non_empty_string(tag)) {
+    zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
+    nr_wordpress_name_the_wt(tag, retval_ptr TSRMLS_CC);
+  }
   nr_php_arg_release(&tag);
+
+  nr_wordpress_handle_tag_stack_after(NR_SPECIALFNPTR_ORIG_ARGS);
 }
 NR_PHP_WRAPPER_END
 
+#endif //OAPI
+
 void nr_wordpress_enable(TSRMLS_D) {
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+  nr_php_wrap_user_function_before_after(NR_PSTR("apply_filters"),
+                            nr_wordpress_apply_filters,
+                            nr_wordpress_apply_filters_after);
+
+  nr_php_wrap_user_function_before_after(NR_PSTR("apply_filters_ref_array"),
+                            nr_wordpress_exec_handle_tag,
+                            nr_wordpress_handle_tag_stack_after);
+
+  nr_php_wrap_user_function_before_after(NR_PSTR("do_action"),
```

Comment:
Would it be possible to add integration tests for `do_action` to exercise `nr_wordpress_exec_handle_tag` and `nr_wordpress_handle_tag_stack_after`?

Reply:
added in [ac44475](https://github.com/newrelic/newrelic-php-agent/pull/559/commits/ac4447541e259e4c6ef19c3bd7e5e45241bb6d34)